src/main/scala/com/thenewmotion/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -80,20 +84,30 @@ class ConnectionActor(
           goto(Disconnected) using NoConnection replying ChannelCreated(child)
       }
 
-    case Event(AmqpShutdownSignal(cause), Connected(connection)) =>
+    case Event(AmqpShutdownSignal(cause), Connected(connection, _)) =>
       if (!cause.isInitiatedByApplication) reconnect(connection)
       goto(Disconnected) using NoConnection
+
+    case Event(blocked: QueueBlocked, data: Connected) =>
```

Comment:
Shouldn't you be setting / unsetting the `connectionBlocked` field in the state here?